### PR TITLE
Fuel balancing (part 1)

### PR DESCRIPTION
**Pull request by mugling**
Vehicle tanks are now consistent with the rule that `vol 1 = 250ml` and the naming of the tanks is improved. `metal_tank_small` is deprecated in favor of the plastic jerrycan which has a near equivalent volume.

| name | size | base item | volume | intended use |
| --- | --- | --- | --- | --- |
| gasoline tank (2L) | 2'000 | metal_tank_little | 8 | scooter |
| gasoline tank (10L) | 10'000 | jerrycan | 40 | bike, buggy |
| gasoline tank (60L) | 60'000 | metal_tank | 240 | car |

Now that vehicles are loaded with realistic quantities of fuel we can next tackle appropriate fuel consumption. A separate PR will deal with `hydrogen` which is currently a special case.

### CODE REVIEW

**Comment by mugling** [comment]
Hydrogen will probably come in 10L pressurised cylinders but this both adjusting recipes/weapons requiring `plasma` and logic to prevent the gas from escaping into the inventory/map tiles

**Comment by Coolthulhu** [comment]
Attaching (welding) a plastic jerrycan to a car sounds really weird to me. Does it happen IRL? Would it be safe?

File: data/json/vehicle_parts.json [hunk 1]
```diff
@@ -3380,34 +3378,33 @@
...
-        "name" : "clean water tank (little)",
+        "name" : "clean water tank (2L)",
         "symbol" : "O",
         "color" : "light_blue",
         "broken_symbol" : "#",
         "broken_color" : "light_blue",
         "damage_modifier" : 80,
         "durability" : 75,
-        "size" : 5,
+        "size" : 10,
```
Comment (Coolthulhu) [inline data/json/vehicle_parts.json]: The item has volume (and capacity) 8.

File: data/json/items/containers.json [hunk 2]
```diff
@@ -475,16 +457,16 @@
...
         "color" : "light_cyan",
-        "name" : "little metal tank",
+        "name" : "metal tank (2L)",
         "category": "other",
-        "description" : "A little metal tank for holding gas or liquids.  Useful for crafting.",
+        "description" : "A small metal tank for gas or liquids.  Useful for crafting.",
         "price" : 1000,
         "weight" : 800,
-        "volume" : 5,
+        "volume" : 8,
```
Comment (Coolthulhu) [inline data/json/items/containers.json]: There are some items which uncraft to little metal tanks. Those should probably be adjusted too, possibly by having them uncraft to something else.

File: data/json/vehicle_parts.json [hunk 3]
```diff
@@ -3085,14 +3084,14 @@
...
-        "name" : "gasoline tank (little)",
+        "name" : "gasoline tank (2L)",
         "symbol" : "O",
         "color" : "light_red",
         "broken_symbol" : "#",
         "broken_color" : "red",
         "damage_modifier" : 80,
         "durability" : 75,
-        "size" : 650,
+        "size" : 2000,
```
Comment (Coolthulhu) [inline data/json/vehicle_parts.json]: Looks like size of a volume 10 item here too.

**Comment by Coolthulhu** [comment]
There is a recipe to craft a big jerrycan out of a small metal tank. I think you missed it, as I don't see it in the diff.

**Comment by mugling** [comment]
> Attaching (welding) a plastic jerrycan to a car sounds really weird to me. Does it happen IRL?

Yes, most vehicle tanks are plastic and attached with metal straps. It's entirely possible to weld plastic but perhaps we should change the installation requirements so you require a welder for the big tank but the smaller ones can use duct tape?

File: data/json/vehicle_parts.json (hunk 1 above)
Comment (mugling) [inline data/json/vehicle_parts.json]: Noted, will fix

File: data/json/vehicle_parts.json (hunk 3 above)
Comment (mugling) [inline data/json/vehicle_parts.json]: Base item should be volume 8

**Comment by mugling** [comment]
> There is a recipe to craft a big jerrycan out of a small metal tank. I think you missed it, as I don't see it in the diff.

Now that doesn't make sense in any case so I'll look into it

**Comment by mugling** [comment]
> There is a recipe to craft a big jerrycan out of a small metal tank. I think you missed it, as I don't see it in the diff.
> 
> Now that doesn't make sense in any case so I'll look into it

Where is this? The recipe I can find uses the bigger variant (`metal_tank`)

**Comment by Coolthulhu** [comment]
Line 3636 of recipe_others.json at my current revision. CTRL+F `"result": "jerrycan_big"` should find it.

**Comment by mugling** [comment]
No worries, a very easy mistake. The nomenclature is terrible which is one of the motivations for this PR

**Comment by Mshock777** [comment]
As @Coolthulhu already said, it just feels weird specifically designed fuel tank with plastic jerrycan, but we will see what everyone else thinks.

**Comment by mugling** [comment]
Most vehicle fuel tanks are plastic and look very much like a plastic jerrycan:

gastank

**Comment by kevingranade** [comment]
Reality check for hydrogen handling, this has come up a number of times.
There are two hydrogen items.
One is compressed hydrogen gas in a tank (albeit not an ordinary one,
hydrogen requires some special handling) that can be created from water
with a craftable electrolysis machine and compressor. This could be used
for fuel cells or a hydrogen cumbustion powered car. This item reflects the
current state if the art compressed hydrogen.
The other is a super-science 'condensed hydrogen' that can only be
contained in an uncraftable tank and can't be created in a player-craftable
machine. This item is usable in high-performance super-science items such
as a micro reactor or a fusion gun.

**Comment by mugling** [comment]
> There are two hydrogen items.

This I did not know.

**Comment by kevingranade** [comment]
At least in principle, I can't recall if anyone has ever bothered to create
the conventional version of hydrogen, it's nearly useless in the context of
dda.  This has come up in the past because people want to add an
electrolysis process to create the super-science version of hydrogen, which
is unreasonable.

**Pull request merged**